Automattic/jetpack
# JS Components: Convert ThemeProvider component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `ThemeProvider` component to TypeScript.
* It also converts the `ThemeProvider` stories to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `ThemeProvider` component work as expected
* Check the consumers of `ThemeProvider` and related components and confirm that they work as expected
* Check for any regression


<img width="1811" alt="Screenshot 2022-08-02 at 11 31 41 AM" src="https://user-images.githubusercontent.com/18226415/182303275-c76c4126-721b-44a9-8278-de5a768f2d1d.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202276314264004
